core/domain/state_domain.py

```diff
@@ -149,6 +150,56 @@ def validate(self, interaction, exp_param_specs_dict):
 
         self.outcome.validate()
 
+    def get_all_html_content_strings(self):
+        """Get all html content strings in the AnswerGroup.
+
+        Returns:
+            list(str). The list of all html content strings in the interaction.
+        """
+        html_list = []
+
+        outcome_html = self.outcome.feedback.html
+        html_list = html_list + [outcome_html]
+        # TODO(#9413): Find a way to include a reference to the interaction
+        # type in the Draft change lists.
+        # See issue: https://github.com/oppia/oppia/issues/9413. We cannot use
+        # the interaction-id from the rules_index_dict until issue-9413 has
+        # been fixed, because this method has no reference to the interaction
+        # type and draft changes use this method. The rules_index_dict below
+        # is used to figure out the assembly of the html in the rulespecs.
+
+        html_field_types_to_rule_specs_dict = json.loads(
+            utils.get_file_contents(
+                feconf.HTML_FIELD_TYPES_TO_RULE_SPECS_FILE_PATH))
+
+        for rule_spec in self.rule_specs:
+            for interaction_and_rule_details in (
+                    html_field_types_to_rule_specs_dict.values()):
+                if (
+                        rule_spec.rule_type in
+                        interaction_and_rule_details['ruleTypes'].keys()):
+                    if interaction_and_rule_details['format'] == 'string':
+                        input_variables = (
+                            html_field_types_to_rule_specs_dict[
+                                'DragAndDropHtmlString']['ruleTypes'][
+                                    rule_spec.rule_type]['htmlInputVariables'])
+                        for input_variable in rule_spec.inputs.keys():
+                            if input_variable in input_variables:
+                                html_list = (
+                                    html_list +
+                                    [rule_spec.inputs[input_variable]])
+                    elif interaction_and_rule_details['format'] == 'set':
+                        if isinstance(rule_spec.inputs['x'], list):
```

Comment:
Also, all of this code should not refer to "x", which is a rule-specific thing. Use the field name given in the dictionary.

Reply:
done